src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/documents/GotenbergApi.kt

```diff
@@ -0,0 +1,32 @@
+package uk.gov.justice.digital.hmpps.managerecallsapi.documents
+
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.http.HttpEntity
+import org.springframework.http.MediaType.MULTIPART_FORM_DATA
+import org.springframework.stereotype.Component
+import org.springframework.util.MultiValueMap
+import org.springframework.web.reactive.function.client.WebClient
+import reactor.core.publisher.Mono
+
+@Component
+class GotenbergApi(
+  private val webClient: WebClient,
+  @Value("\${gotenberg.endpoint.url}") private val gotenbergEndpointUrl: String
+) {
+  fun merge(mergeRequest: MultiValueMap<String, HttpEntity<*>>): Mono<ByteArray> =
+    gotenbergResponse("/merge", mergeRequest)
+
+  fun convertHtml(convertHtmlRequest: MultiValueMap<String, HttpEntity<*>>): Mono<ByteArray> =
+    gotenbergResponse("/convert/html", convertHtmlRequest)
+
+  private fun gotenbergResponse(
```

Comment:
Not a biggie, but I find that `gotenbergResponse` doesn't describe very well what the function is doing. How about something like `callGotenberg`?

Reply:
I think it read nicer when it wasn't an expression body.
`return gotenbergResponse()`
although maybe something like this is better:
```
fun convertHtml() = gotenbergResponseFor(path, body)
```
Depends how you like your code to read really